### PR TITLE
Kvm health implementation

### PR DESCRIPTION
**Pull request by tfussell**
This PR adds a mock `KVMConfig` as returned by `kubectl get kvmconfig <cluster_id> -o json` and a simple red/yellow/green health check based on the number of nodes. It also includes a Response struct that we can iterate on as we work on health logic. There's lots more to add here, but I think this is enough for a first step.

Towards https://github.com/giantswarm/giantswarm/issues/6893

### CODE REVIEW

File: server/endpoint/health/searcher/response.go [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+	Health       string `json:"health"`
+	Creating     bool   `json:"creating"`
+	Upgrading    bool   `json:"upgrading"`
+	Deleting     bool   `json:"deleting"`
+	Normal       bool   `json:"normal"`
+	NodeCount    int    `json:"nodeCount"`
+	MaxNodeCount int    `json:"maxNodeCount"`
+}
+
+type NodeStatus struct {
```
Comment (MarcelMue) [inline server/endpoint/health/searcher/response.go]: There should be more structure here. E.g.
```
NodeStatus
    Limit
        CPU
```
And so on.
That produces more structs but makes it easier to change specific parts and extend the structure.

File: mock/kvm.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
+package mock
```
Comment (MarcelMue) [inline mock/kvm.go]: I like the idea of this package.

File: server/endpoint/health/searcher/response.go (hunk 1 above)
Comment (tfussell) [reply server/endpoint/health/searcher/response.go]: That makes sense. I've cleaned this up now.

File: server/endpoint/health/searcher/response.go [hunk 3]
```diff
@@ -1,5 +1,48 @@
 package searcher
 
+// GeneralStatus holds data about the overall health/status of a cluster
+type GeneralStatus struct {
+	Health       string `json:"health"`
+	Creating     bool   `json:"creating"`
+	Upgrading    bool   `json:"upgrading"`
+	Deleting     bool   `json:"deleting"`
+	Normal       bool   `json:"normal"`
+	NodeCount    int    `json:"nodeCount"`
```
Comment (MarcelMue) [inline server/endpoint/health/searcher/response.go]:
```suggestion
	NodeCount    int    `json:"node_count"`
```
We usually camel case like this in our other APIs when json is involved - I am going to be honest though, no idea why.

**Review by MarcelMue — APPROVED** [review]
Just some nitpicks - LGTM otherwise.

File: server/endpoint/health/searcher/response.go (hunk 3 above)
Comment (tfussell) [reply server/endpoint/health/searcher/response.go]: Hmm, a little Python flair in our Go? Fixed.

**Pull request merged**